searx/plugins/rerank.py

```diff
@@ -0,0 +1,77 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+"""Plugin which reranks the search results using the Okapi BM25 algorithm.
+Before enabling the Rerank plugin, you must the install the pip package ``bm25s``.
+
+Enable in ``settings.yml``:
+
+.. code:: yaml
+
+  enabled_plugins:
+    ..
+    - 'Rerank plugin'
+
+By default, the engine list is retained, so results found by multiple engines receive a score boost.
+The following setting can be used to ensure that the engine list only contains the first engine.
+This will prevent overlapping search engine results from affecting the ranking:
+
+.. code:: yaml
+
+  rerank:
+    remove_extra_engines: true
+
+"""
+
+from searx import settings
+
+try:
+    import bm25s
+except ImportError:
+    # Import error is ignored because the admin has to install bm25s manually to use the engine
+    pass
+
+name = 'Rerank plugin'
+description = 'Rerank search results, ignoring original engine ranking'
+default_on = False
+preference_section = 'general'
```

Comment:
It's also visible in the settings ui when `bm25s` is not installed. We should however hide the plugin in that case.

Could be implemented like here: https://github.com/searxng/searxng/pull/3377/files#diff-554a7ca4c7612fcaa88c3e01928cdea683ae55d3a7c35cbc2677539a1001710aR41 and https://github.com/searxng/searxng/pull/3377/files#diff-340f925dc468a2e8194cc4059868c642ad96dadac281836d0969074b944e85e3R87.
(We removed that `is_allowed` method recently since it's not been used anywhere, so it'd need to be re-added)